compiler/rustc_codegen_llvm/src/back/write.rs

```diff
@@ -523,6 +523,12 @@ pub(crate) unsafe fn optimize(
     let module_name = module.name.clone();
     let module_name = Some(&module_name[..]);
 
+    if let Some(false) = config.new_llvm_pass_manager {
+        diag_handler.warn(
+            "ignoring `-Z new-llvm-pass-manager=no`, which is no longer supported with LLVM 15",
```

Comment:
This message isn't actually gated on LLVM 15 here though?

Reply:
Yes ... we could make it a general deprecation, but with the message as written, this should indeed have a version check. Thanks!